viya4-iac-aws-darksite/README.md

```diff
@@ -0,0 +1,39 @@
+
+### Procedures
+
+1.  **Build a Private VPC for EKS:**
+
+    - **Note:** _Creating AWS VPCs for an EKS cluster can be accomplished in many different ways. Some of those methods include using a CloudFormation template, the AWS console, the AWS CLI, or Terraform scripts._
+
+    - **Note**: If you have an existing VPC environment consisting of a private VPC and a public VPC (optional) that constitutes a Dark Site configuration, skip to the **_Create Custom AMI for Jumpserver/NFSServer_** in step #2 below.
+
+    - Background Information - AWS Reference documentation with details on how to create a VPC for any EKS cluster: https://docs.aws.amazon.com/eks/latest/userguide/creating-a-vpc.html
+
+    - For the cluster VPC ranges, `viya4-iac-aws` defaults to using a CIDR of /16. The `viya4-iac-aws` project creates a /21 VPC with both /22 and /28 sized subnets.  Excluding the `control_plane` subnets, the sizes were chosen by estimating the number of pods (doubling that number to account for viya updates), services, AWS overhead, and then adding a few hundred as a buffer.  
+
+2.  **Create Custom AMI for Jumpserver/NFSServer**
+    - The standard base AMI image used by `viya4-iac-aws` does not include the required NFS related Linux distributution packages.  Normally, `viya4-iac-aws` will attempt to install the NFS packages as part of the VM initialization.  In a Dark Site without access to Internet based resources, installation from an Internet based repository will not be possible.  To mitigate that issue, we'll need to create a custom AMI and then modify the local copy of our `viya4-iac-aws` repository to add references to that custom AMI as well as remove some of the initialization steps in the cloud-init files (for jumpserver and nfs-server). 
+
+    - Instructions to complete the custom AMI creation steps can be found [here](https://github.com/sassoftware/viya4-iac-aws/tree/feat/iac-1117/viya4-iac-aws-darksite/custom-ami/) [here](https://github.com/sassoftware/viya4-iac-aws/tree/main/viya4-aws-darksite/custom-ami/).
+
+3.  **Build tfvars:**
+    - The Terraform scripts create the necessary BYON subnet_ids, `viya4-iac-aws` can be configured to use those subnets for all the subnet_ids map BYON variables in the tfvars file (public, private, control_plane and database).
+    - Do not specify any public access cidrs in your tfvars file.
+    - Set `cluster_api_mode` to private.
+
+4.  **Deploy viya4-iac-aws:**
+    - Ensure that the deployment machine has a route and is allowed ingress to the cluster control plane before attempting a deployment. Executing a `kubectl` command such as `kubectl get nodes` from the deployment VM in the public VPC can be used to confirm connectivty to the EKS control plane.
```

Comment:
I'm confused here, perhaps some of the terminology being used isn't helping?
How can you confirm connectivity to the control plane before building the cluster?  You don't have the URI for the API server UNTIL you have a cluster.  
Of course if we are talking about later, on the Viya 4 Deployment machine, that's different, but I don't think we are talking about that here.  Some terminology changes could help:

Terminology
Typically we use "deploy" for SAS Viya and the viya4-deployment project.
For IaC, I find it's clearer to stick to terms like "build" the infrastructure.  The machine where IaC runs is then the execution or build machine.
Ingress:  while valid to use ingress as you are here, it's easily confused with the all important topic of "cluster ingress and the ingress controller".  I find it better to stick to "inbound access" when talking about "allowed" routes, since we are talking about security groups now, and AWS uses "INBOUND rules" in this context.